src/niswitch/metadata/functions_addon.py

```diff
@@ -0,0 +1,75 @@
+# These dictionaries are applied to the generated functions dictionary at build time
+# Any changes to the API should be made here. functions.py is code generated
+
+# By default all functions in functions.py will be generated as a public function
+# This will override that with private - add '_' to the beginning of the name, or
+# don't generate at all
+functions_codegen_method = {
+    'InitWithOptions':  { 'codegen_method': 'private',  },
+    'Initiate':         { 'codegen_method': 'private',  },
+    'close':            { 'codegen_method': 'private',  },
+    'Abort':            { 'codegen_method': 'private',  },
+    'CheckAttribute.+': { 'codegen_method': 'no',       },  # We do not include any Check Attribute functions
+    '.etAttribute.+':   { 'codegen_method': 'private',  },  # All Set/Get Attribute functions are private
+    'init':             { 'codegen_method': 'no',       },
+    'GetError':         { 'codegen_method': 'private',  },
+    'GetErrorMessage':  { 'codegen_method': 'private',  },
+    'ClearError':       { 'codegen_method': 'private',  },
+    'LockSession':      { 'codegen_method': 'private',  },
+    'UnlockSession':    { 'codegen_method': 'private',  },
+    'UnlockSession':    { 'codegen_method': 'private',  },
+}
+
+# Attach the given parameter to the given enum from enums.py
+functions_enums = {
+    'GetRelayPosition':             { 'parameters': { 2: { 'enum': 'RelayPosition',                                        }, }, },
+    'RelayControl':                 { 'parameters': { 2: { 'enum': 'RelayAction',                                          }, }, },
+    'CanConnect':                   { 'parameters': { 3: { 'enum': 'PathCapability',                                       }, }, },
+    'RouteScanAdvancedOutput':      { 'parameters': { 1: { 'enum': 'TriggerInputConnector',                                }, 
+                                                      2: { 'enum': 'TriggerInputBusLine',                                  }, }, },
+    'RouteTriggerInput':            { 'parameters': { 1: { 'enum': 'TriggerInputConnector',                                }, 
+                                                      2: { 'enum': 'TriggerInputBusLine',                                  }, }, },
+    'ConfigureScanList':            { 'parameters': { 2: { 'enum': 'ScanMode',                                             }, }, },
+    'ConfigureScanTrigger':         { 'parameters': { 2: { 'enum': 'TriggerInputConfigureScanTrigger',                     }, 
+                                                      3: { 'enum': 'ScanAdvancedOutputConfigureScanTrigger',               }, }, },
+}
+
+# TODO(texasaggie97) can we get rid of this now that we are code generating the ivi-dance method of buffer retrieval?
```

Comment:
What is this comment about, does it still apply?

Reply:
This is also in the DMM function_addon, if I recall correctly, not having the params_types causes various things to break

Reply:
Seems like something we need to investigate.
Please open an issue and then reference it in the TODOs.

Reply:
Created #259